### ISSUE TITLE
Random LIT tests failures

### ISSUE THREAD

**Issue by shubhamnarlawar** [ballerina-platform/nballerina#175]
**Description:**
After PR #163 got merged in the master, there seems some random LIT tests are failing whenever a new PR is sent to master i.e. 

However, after re-running the jobs, it is passing.

**Steps to reproduce:**
1. Send a pull request to master.
2. If the github action check is successful, then re-run the jobs 1 or 2 times untill there are failing LIT tests.

**OS, DB, other environment details and versions:**
Operating System
  Ubuntu
  20.04.2
  LTS

**Comment by shubhamnarlawar** [comment]
@Kishanthan and @ruvi-d - Should we revert #175 ?

**Comment by shubhamnarlawar** [comment]
There seems an issue in the github workflow file.

**Comment by shubhamnarlawar** [comment]
> There seems an issue in the github workflow file which comes in picture after #175 got merged in the master.

The regression is caused by #175

**Comment by shubhamnarlawar** [comment]
I have ported LIT tests to use print routines from rust instead of `$?` - 

1. Based on the function flags, if the function is external (for example - `printu32()` in rust), function body is not generated. Instead there is function call.

2. After making above change, I have added print routines from rust in all the bal testcases. But some random tests are failing with below error - (Some times all tests are passing).
`error: cannot find `-lballerina_rt`

3. The only codegen logic I have used here is - if the function is external then the function body is not emitted.

@ayonam - Can you please take a quick look at the code to check if any part of code seems to cause above failure and give suggestions?
https://github.com/ballerina-platform/nballerina/pull/163/files

**Comment by ruvi-d** [comment]
I think I found the issue.
In the current test runner, we are piping the standard errors from all the tools to the same file `err.log`. Since Ballerina is actually outputting an error due to the missing external function, the err.log is not empty and the subsequent build steps fail. The following patch will fix the issue. 
https://github.com/ruvi-d/nballerina/commit/e106cbe55334e740ce5888c44fea486aabbab67c

Please send a new PR with this change.

**Comment by shubhamnarlawar** [comment]
> I think I found the issue.
> In the current test runner, we are piping the standard errors from all the tools to the same file `err.log`. Since Ballerina is actually outputting an error due to the missing external function, the err.log is not empty and the subsequent build steps fail. The following patch will fix the issue.
> [ruvi-d@e106cbe](https://github.com/ruvi-d/nballerina/commit/e106cbe55334e740ce5888c44fea486aabbab67c)
> 
> Please send a new PR with this change.

Done. I have sent PR #177 along with above fix.
Thanks @ruvi-d
